rust/processor/src/db/postgres/migrations/2024-08-08-024053_emojicoin_events/up.sql

```diff
@@ -0,0 +1,299 @@
+  last_swap_is_sell BOOLEAN NOT NULL,
+  last_swap_avg_execution_price_q64 NUMERIC NOT NULL,
+  last_swap_base_volume BIGINT NOT NULL,
+  last_swap_quote_volume BIGINT NOT NULL,
+  last_swap_nonce BIGINT NOT NULL,
+  last_swap_time TIMESTAMP NOT NULL,
+
+  PRIMARY KEY (market_id, market_nonce)
+);
+
+CREATE TABLE liquidity_events (
+  -- Transaction metadata.
+  transaction_version BIGINT NOT NULL,
+  sender VARCHAR(66) NOT NULL,
+  entry_function VARCHAR(200),
+  transaction_timestamp TIMESTAMP NOT NULL,
+  inserted_at TIMESTAMP NOT NULL DEFAULT NOW(),
+
+  -- Market and state metadata.
+  market_id BIGINT NOT NULL,
+  symbol_bytes BYTEA NOT NULL,
+  bump_time TIMESTAMP NOT NULL,
+  market_nonce BIGINT NOT NULL,
+  trigger triggers NOT NULL,
+
+  -- Liquidity event data.
+  provider VARCHAR(66) NOT NULL,
+  base_amount BIGINT NOT NULL,
+  quote_amount BIGINT NOT NULL,
+  lp_coin_amount BIGINT NOT NULL,
+  liquidity_provided BOOLEAN NOT NULL,
+  pro_rata_base_donation_claim_amount BIGINT NOT NULL,
+  pro_rata_quote_donation_claim_amount BIGINT NOT NULL,
+
+  -- State event data.
+  clamm_virtual_reserves_base BIGINT NOT NULL,
+  clamm_virtual_reserves_quote BIGINT NOT NULL,
+  cpamm_real_reserves_base BIGINT NOT NULL,
+  cpamm_real_reserves_quote BIGINT NOT NULL,
+  lp_coin_supply NUMERIC NOT NULL,
+  cumulative_stats_base_volume NUMERIC NOT NULL,
+  cumulative_stats_quote_volume NUMERIC NOT NULL,
+  cumulative_stats_integrator_fees NUMERIC NOT NULL,
+  cumulative_stats_pool_fees_base NUMERIC NOT NULL,
+  cumulative_stats_pool_fees_quote NUMERIC NOT NULL,
+  cumulative_stats_n_swaps BIGINT NOT NULL,
+  cumulative_stats_n_chat_messages BIGINT NOT NULL,
+  instantaneous_stats_total_quote_locked BIGINT NOT NULL,
+  instantaneous_stats_total_value_locked NUMERIC NOT NULL,
+  instantaneous_stats_market_cap NUMERIC NOT NULL,
+  instantaneous_stats_fully_diluted_value NUMERIC NOT NULL,
+  last_swap_is_sell BOOLEAN NOT NULL,
+  last_swap_avg_execution_price_q64 NUMERIC NOT NULL,
+  last_swap_base_volume BIGINT NOT NULL,
+  last_swap_quote_volume BIGINT NOT NULL,
+  last_swap_nonce BIGINT NOT NULL,
+  last_swap_time TIMESTAMP NOT NULL,
+
+  PRIMARY KEY (market_id, market_nonce)
+);
+
+-------------------------------------------------------------------------------
+--
+--                                  Indexes
+--
+-------------------------------------------------------------------------------
+
+-- Querying the swap events for a market, descending chronologically.
+CREATE INDEX swaps_by_mkt_and_time_idx
+ON swap_events (market_id, market_nonce DESC);
+
+-- Querying the chat events for a market, descending chronologically.
+CREATE INDEX chats_by_mkt_and_time_idx
+ON chat_events (market_id, market_nonce DESC);
+
+-- Querying the candlestick data for a market by its period,
+-- descending chronologically by the candlestick open time.
+CREATE UNIQUE INDEX prdc_evts_by_res_idx
+ON periodic_state_events (market_id, period, start_time DESC);
```

Comment:
Do we want `DESC` here ? I feel like we would query from 7 days ago to now, not frow now to 7 days ago.

Reply:
I believe I did it this way because the charting library dynamically requests to fetch data from the endpoint based on how many candlesticks are returned. Essentially it doesn't ask for "all candlesticks in the last day" (which is what could be ordered by `start_time ASC`) but it asks for candlesticks repeatedly until it's gone back far enough to be satisfied.

I don't know if that's actually a proper reason to not do it but I remember fighting with it in the past and that there was a reason I did it that way (because I agree, intuitively you'd think it'd be the other way around)

Reply:
Basically, this index would work for `SELECT * FROM table ORDER BY start_time DESC LIMIT 100` (ie get the last 100 candlesticks) but not for `SELECT * FROM table WHERE start_time > X ORDER BY start_time LIMIT 100` (get the first 100 candlesticks after date X). You should choose depending on which is more appropriate here.

Reply:
```json
{
  "pathname": "//inbox_periodic_states",
  "search": "?select=*&market_id=eq.582&period=eq.300000000&start_time=gte.1723546800000000&offset=0&limit=100&order=start_time.asc",
  "searchParams": {
    "select": "*",
    "market_id": "eq.582",
    "period": "eq.300000000",
    "start_time": "gte.1723546800000000",
    "offset": "0",
    "limit": "100",
    "order": "start_time.asc"
  }
}
```

So the above is the actual search params the query is doing currently

But it does that multiple times, by exhaustively querying from a `start: Date` to an `end: Date`

It only stops when the query returns less than `LIMIT` (aka 100) results, indicating it's gotten everything from `start` to `end`

For now I'll do:

`ON periodic_state_events (market_id, period, start_time ASC);`

We can change it later too, just want to get this in for now

Reply:
Addressed here: https://github.com/econia-labs/aptos-indexer-processors/pull/34/commits/d65d01b0a7b70592e5b897ec39669b7af45bbe92